bsip-0073.md

```diff
@@ -0,0 +1,210 @@
+to an old price expired, or due to asset options changed, if the new feed
+price is valid, and if the new feed price in the direction of "X debt asset
+per collateral asset" is higher than the old feed price or the old feed
+price was invalid,
+* current logic doesn't handle settle orders,
+* after the hard fork, the new logic would be:
+  * if there are limit orders selling collateral asset below the new feed
+    price, and there are settle orders whose `fill_asap` is `true`,
+    match the settle orders with the limit orders, in other words, treat
+    those settle orders as taker limit orders buying at the new feed price.
+
+### Do not update `force_settled_volume` before the delay
+
+When a settle order is matched and (partially) filled before the delay, 
+it doesn't affect `force_settled_volume` (which indicates how much debt
+asset has been force-settled in current maintenance interval).
+
+## Processing settle orders after the delay
+
+Currently, when filling a settle order after the delay (note: it implies
+some conditions E.G. price feed is valid and total settled volume in current
+maitenance interval doesn't exceed maximum allowed volume),
+the settle order will be matched against the debt position with the least
+collateral ratio, the fill price in the direction of "X debt asset
+per collateral asset" would be
+`fill_price = feed_price * (1 + foce_settlement_offset)`.
+
+After the hard fork, when processing a settle order after the delay,
+* firstly try to match it with the margin calls and limit orders selling
+  below `fill_price`, in other words, treat it as a taker limit order buying
+  at `fill_price`.
+  Note: this step doesn't affect `force_settled_volume`.
+* if the settle order still exists, process it with the logic before the
+  hard fork.
+  Note: this step does affect `force_settled_volume` as before.
+
+## API
+
+APIs which return `settle_order_object` need to return the new `fill_asap`
+field.
+
+APIs which return combined order book can combine settle orders whose
+`fill_asap` is true with limit orders in the same direction.
+
+## CLI
+
+There was a `settle_asset` command in CLI.
+
+Need to add a new command E.G. `settle_asset_ext` in CLI so that users would
+be able to create force-settlement orders with the new `fill_asap` option.
+
+## GUI/UX
+
+The new `fill_asap` option need to be presented and can be used in UI after
+the hard fork.
+
+When there are settle orders with `fill_asap` set to `true`, UI can show them
+as special buy orders which are buying at feed price in the order book.
+
+# Discussion
+
+With this BSIP, we provided a tool that can be used by debt asset holders to
+convert their debt asset to collateral asset more conveniently and flexibly.
+However, it's not guaranteed that a settle order will be filled at better
+price if the owner chose to fill it "as soon as possible". Market paticipants
+should always make their own decisions on whether to use the new tool.
+
```

Comment:
Please add these discussion points:

* From the perspective of the chain it is more desirable to execute settlements against margin calls than against market orders, because it improves the collateral ratio of these. In that sense it contradicts the goals of https://github.com/bitshares/bsips/pull/193 .
* The proposed functionality is mostly convenience for users. Manual selling on the market is (almost) always available when it would be used automatically by this proposal (exception: new incoming limit orders).
* The proposed functionality complicates the market engine and possibly the get_order_book call (see https://github.com/bitshares/bitshares-core/issues/1958). It is probably harmful for performance in the long run, which is particularly undesirable for functionality that can mostly be emulated client-side.

Reply:
> From the perspective of the chain it is more desirable to execute settlements against margin calls than against market orders

I think this is not really from the perspective of the chain, but rather from individual asset owners for the assets' performance, although some MPAs in BitShares are owned by committee-account.

IMHO the UX improvement is more important.

#193 is mostly about positions whose CR < MCR. For this BSIP,
* if no debt position's CR is below MCR, and there are limit orders selling below `feed_price * (1+offset)`, IMHO it doesn't make sense to force the debt position to increase CR;
* if a debt position's CR is below MCR,
  * if no limit order is selling below MSSP, the debt position would be matched and filled at MSSP (but not at `feed_price*(1+offset)` as pre-HF), which would likely increase CR of the position;
  * if there are limit orders selling below MSSP which caused margin calls unable to be filled, I think a bidding mechanism is better for increasing overall CR.